### DISCUSSION TITLE
Calculation of strain energy density with the eigen strains

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#24853] [Q&A General] [answered]
Hello MOOSE experts,

I am trying to add 2 way coupling from the paper as shown in attached figure.

I found one material kernel which can be used to evaluate the strain energy density but this is without eigen strains. I wonder if there is any material kernel to calculate the strain energy density with the eigen strains. Also, I want to calculate it for finite deformation problem.

https://mooseframework.inl.gov/source/materials/StrainEnergyDensity.html

![image](https://github.com/idaholab/moose/assets/94848840/a283cfb1-f41c-4123-85f8-fff34b023525)

Link to the article is https://www.sciencedirect.com/science/article/pii/S0378775316304074.

Thanks,
Avtar

**Comment by GiudGiud** [comment]
@recuero @vanwdani

**Reply by recuero** [reply]
Strain energy density calculations should not include eigenstrains as these do not generate stresses. In this sense, the existing MOOSE object should do the job (it even seems that the equation you posted uses a single type of strain, with the subscript "e"; maybe for elastic).

**Reply by avtarsinghh1991** [reply]
Thanks @recuero. Does this strain energy density calculation involves only elastic strain or the total strain? I want to calculate the strain energy density only from the elastic part. Also, is it compatible with large deformation settings?

**Reply by recuero** [reply, marked as answer]
MOOSE computes the strain energy density using the elastic strain (if you look in the code it may not be called exactly that, e.g. `_strain_increment`, but it's the elastic strain). Yes, it is compatible with large deformation kinematics.
